### PR TITLE
feat(search): moved result formatting logic into views

### PR DESCRIPTION
**Pull request by jdalsem**
- [x] rss view
- [x] rename comment hook

### CODE REVIEW

File: mod/search/views/default/search/elements/content.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Default view for content used in search results
+ *
+ * Display largely controlled by a set of overrideable volatile data:
+ *   - search_matched_title (defaults to entity->getDisplay())
```
Comment (jdalsem) [inline mod/search/views/default/search/elements/content.php:6]: incorrect

File: mod/search/views/default/search/elements/icon.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+if ($entity->hasIcon($size) || $entity instanceof \ElggFile) {
+	echo elgg_view_entity_icon($entity, $size);
+} else if ($entity->getType() === 'user' || $entity->getType() === 'group') {
+	echo elgg_view_entity_icon($entity, $size);
+} else if ($owner instanceof ElggUser) {
+	echo elgg_view_entity_icon($owner, $size);
+} else if ($entity->getContainerEntity() instanceof ElggUser) {
+	// display a generic icon if no owner, though there will probably be
+	// other problems if the owner can't be found.
+	echo elgg_view_entity_icon($entity, $size);
```
Comment (jdalsem) [inline mod/search/views/default/search/elements/icon.php:32]: is this intended? should it not be container entity?

File: mod/search/views/default/search/elements/icon.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * @uses $vars['entity']    Entity returned in a search
+ * @uses $vars['icon_size'] Size of icon (default 'small')
+ */
+$entity = elgg_extract('entity', $vars);
+if (!$entity instanceof ElggEntity) {
+	return;
+}
+
+if ($entity->getVolatileData('search_icon')) {
+	return $entity->getVolatileData('search_icon');
```
Comment (jeabakker) [inline mod/search/views/default/search/elements/icon.php:17]: shouldn't this be a `echo` and then `return`

File: mod/search/views/default/search/elements/icon.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+$size = elgg_extract('icon_size', $vars, 'small');
+
+if ($entity->hasIcon($size) || $entity instanceof \ElggFile) {
+	echo elgg_view_entity_icon($entity, $size);
+} else if ($entity->getType() === 'user' || $entity->getType() === 'group') {
+	echo elgg_view_entity_icon($entity, $size);
+} else if ($owner instanceof ElggUser) {
+	echo elgg_view_entity_icon($owner, $size);
+} else if ($entity->getContainerEntity() instanceof ElggUser) {
```
Comment (jeabakker) [inline mod/search/views/default/search/elements/icon.php:30]: why can't the container be a `group`?

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: mod/search/views/default/search/elements/icon.php (hunk 4 above)
Comment (jdalsem) [reply mod/search/views/default/search/elements/icon.php:31]: it can but is was not

File: mod/search/views/default/search/elements/icon.php (hunk 2 above)
Comment (jdalsem) [reply mod/search/views/default/search/elements/icon.php:32]: nope...

**Review by jeabakker — APPROVED** [review]

**Pull request merged**